### PR TITLE
Bug: interpolate.NearestNDInterpolator (issue #10352)

### LINKED ISSUE #10352: BUG: interpolate.NearestNDInterpolator with pandas
interpolate.NearestNDInterpolator does not work as expected when used  with selected pandas dataframe.
This is due to the index being maintained when making selections in pandas.

### Reproducing code example:
```
import numpy as np
import pandas as pd
from scipy import interpolate

df = pd.DataFrame(np.array([[0, 0, 0, 0, 1, 0, 1],
                            [0, 0, 0, 0, 0, 1, 1],
                            [0, 0, 0, 0, 1, 1, 2]]).T, columns=['x', 'y', 'z'])
df_select = df[3:]
NI = interpolate.NearestNDInterpolator((df_select.x, df_select.y), df_select.z)
print(NI([0.1, 0.9], [0.1, 0.9]))
```
I expect [0, 2] to be output.
But output is [Nan, 0] as pandas.Series.

This is due to the index being maintained when making selections in pandas.
Specifically, `df_select.z` has index[3, 4, 5, 6].
But, self.tree.query (xi) line 81, in scipy/interpolate/ndgriddata.py returns a index that assumes that the index starts from zero.
So, self.tree.query (xi) return [0, 3]
Therefore, self.values[i] line 82, in scipy/interpolate/ndgriddata.py using Invalid index.

### Note
if case of
```
df_select = df[3:].reset_index()
```
or
```
NI = interpolate.NearestNDInterpolator((df_select.x, df_select.y), np.array(df_select.z))
```
it works as expected.

Also, this bug does not occur in interpolate.LinearNDInterpolator.

### Scipy/Numpy/Python version information:
```
1.3.0 1.16.4 sys.version_info(major=3, minor=6, micro=8, releaselevel='final', serial=0)
```

### PR DESCRIPTION
**Pull request by yamaken1343**
fix  #10352
The problem is solved by casting the searched array.

### CODE REVIEW

File: scipy/interpolate/ndgriddata.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def __init__(self, x, y, rescale=False, tree_options=None):
         if tree_options is None:
             tree_options = dict()
         self.tree = cKDTree(self.points, **tree_options)
-        self.values = y
+        self.values = np.array(y)
```
Comment (pv) [inline scipy/interpolate/ndgriddata.py]: np.asarray(y)

**Review by pv — CHANGES_REQUESTED** [review]
Small change needed.

Test? (If you can use a Python list instead of pandas dataframe.)

File: scipy/interpolate/ndgriddata.py (hunk 1 above)
Comment (yamaken1343) [reply scipy/interpolate/ndgriddata.py]: Thank you @pv. 
I think that the better performance.

**Comment by yamaken1343** [comment]
The sugestion was reflected.
And added a test. But I was not able to test using pandas.
When import pandas in `test_nddata.py`, `E   RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility.` occard.

File: scipy/interpolate/tests/test_ndgriddata.py [hunk 2]
```diff
@@ -175,3 +175,17 @@ def test_nearest_options():
     nndi_o = NearestNDInterpolator(x, y, tree_options=opts)
     assert_allclose(nndi(x), nndi_o(x), atol=1e-14)
 
+
+def test_nearest_nd_interpolator():
```
Comment (pv) [inline scipy/interpolate/tests/test_ndgriddata.py]: This should be named `test_nearest_list_argument`.

**Comment by pv** [comment]
thanks, merged

**Comment by yamaken1343** [comment]
thank you review and so on.

**Pull request merged**